forge/src/coverage.rs

```diff
@@ -0,0 +1,224 @@
+use comfy_table::{Attribute, Cell, Color, Row, Table};
+pub use foundry_evm::coverage::*;
+use std::{collections::HashMap, io::Write, path::PathBuf};
+
+/// A coverage reporter.
+pub trait CoverageReporter {
+    fn build(&mut self, map: CoverageMap);
+    fn finalize(self) -> eyre::Result<()>;
+}
+
+/// A simple summary reporter that prints the coverage results in a table.
+pub struct SummaryReporter {
+    /// The summary table.
+    table: Table,
+    /// The total coverage of the entire project.
+    total: CoverageSummary,
+}
```

Comment:
the list of reporters may grow and become more complex, so I think we should consider moving all of them to their own module